sympy/core/tests/test_assumptions.py

```diff
@@ -1154,9 +1167,14 @@ def test_issue_10302():
     r = Symbol('r', real=True)
     u = -(3*2**pi)**(1/pi) + 2*3**(1/pi)
     i = u + u*I
+
     assert i.is_real is None  # w/o simplification this should fail
     assert (u + i).is_zero is None
-    assert (1 + i).is_zero is False
+
+    # XXX: Ideally (1 + i).is_zero would give False.
+    # It fails due to a bug in numerical evaluation.
+    assert (1 + i).is_zero is None
```

Comment:
I think the problem is this:
```python
In [4]: (1 + i).n(2)
Out[4]: 1.0 + 0.e-22⋅ⅈ

In [5]: (1 + i)._eval_evalf(2)
Out[5]: 0.e+0
```
There are various places that call `_eval_evalf(2)`.

Reply:
`pure_complex` should be used in such cases, shouldn't it?

Reply:
What we have is this:
```python
In [6]:     u = -(3*2**pi)**(1/pi) + 2*3**(1/pi)
   ...:     i = u + u*I

In [7]: 1 + i
Out[7]: 
        1                ⎛        1       ⎞
        ─          1     ⎜        ─      1⎟
        π          ─     ⎜        π      ─⎟
  ⎛   π⎞           π     ⎜  ⎛   π⎞       π⎟
- ⎝3⋅2 ⎠  + 1 + 2⋅3  + ⅈ⋅⎝- ⎝3⋅2 ⎠  + 2⋅3 ⎠

In [8]: from sympy.core.evalf import pure_complex

In [9]: print(pure_complex(1+i))
None
```

Reply:
yes - that is not a "pure complex" sort of number that can be trivially determined to be real or imaginary.

Reply:
With the change I suggested above this is:
```python
>>> u = -(3*2**pi)**(1/pi) + 2*3**(1/pi)
>>> i = u + u*I
>>> (1+i).is_zero
False
```